Sources/HummingbirdWebSocket/WebSocketOutboundWriter.swift

```diff
@@ -50,4 +50,59 @@ public struct WebSocketOutboundWriter: Sendable {
             try await self.handler.write(frame: frame)
         }
     }
+
+    struct MessageWriter {
+        let handler: WebSocketHandler
+        var prevFrame: WebSocketFrame?
+        mutating func write(_ data: ByteBuffer, opcode: WebSocketOpcode) async throws {
+            if let prevFrame {
+                try await self.handler.write(frame: prevFrame)
+                self.prevFrame = .init(fin: false, opcode: .continuation, data: data)
+            } else {
+                self.prevFrame = .init(fin: false, opcode: opcode, data: data)
+            }
+        }
+
+        func finish() async throws {
+            if var prevFrame {
+                prevFrame.fin = true
+                try await self.handler.write(frame: prevFrame)
+            }
+        }
+    }
+
+    /// Write a single WebSocket text message as a series of fragmented frames
+    /// - Parameter write: Function writing frames
+    public func withTextMessageWriter<Value>(_ write: ((String) async throws -> Void) async throws -> Value) async throws -> Value {
```

Comment:
Can we create a type for the writer, that is `callableAsFunction`? I like that pattern here, because we can then add new overloads on that function call. That would allow us to send `Data` or `ByteBuffer` payloads with a `.text` opcode - helpful for JSONEncoder output.

Reply:
done